### PR TITLE
Add tests for Message

### PR DESCRIPTION
**Pull request by donny-dont**
This adds tests for the `Message` class and begins the migration from `unittest` to `test`.

### CODE REVIEW

**Comment by donny-dont** [comment]
@nex3 the tests were straight up copied from `shelf` with modifications to account for any differences between `http` and `shelf`. Also added expectations for the value of `Message.isEmpty` which weren't present in `shelf`. If any aren't applicable I'll remove.

Only other question was around `contentLength`. So in `shelf` it'll automatically get set to `0` if body is `null`. So wondering if we need to do anything to `Message.isEmpty`.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+  path: '>=0.9.0 <2.0.0'
+  stack_trace: '>=0.9.1 <2.0.0'
 dev_dependencies:
-  unittest: ">=0.9.0 <0.12.0"
-environment:
-  sdk: ">=1.24.0-dev.0.0 <2.0.0"
+  test: '^0.12.18'
+# Override dependency on package_resolver to enable test
+dependency_overrides:
+  package_resolver: '^1.0.0'
```
Comment (nex3) [inline pubspec.yaml]: What's with all the formatting changes?

**Comment by donny-dont** [comment]
```
  /// If `true`, the stream returned by [read] won't emit any bytes.
  ///
  /// This may have false negatives, but it won't have false positives.
  bool get isEmpty => _body.contentLength == 0;
```

There isn't currently a check for `null` its just checking for length being 0.

File: pubspec.yaml (hunk 1 above)
Comment (donny-dont) [inline pubspec.yaml]: I can just move it all back to using `"`

Comment (donny-dont) [inline pubspec.yaml]: Reverted

File: pubspec.yaml [hunk 2]
```diff
@@ -10,6 +10,9 @@ dependencies:
   path: ">=0.9.0 <2.0.0"
   stack_trace: ">=0.9.1 <2.0.0"
 dev_dependencies:
-  unittest: ">=0.9.0 <0.12.0"
+  test: "^0.12.18"
+# Override dependency on package_resolver to enable test
+dependency_overrides:
+  package_resolver: '^1.0.0'
```
Comment (nex3) [inline pubspec.yaml]: It might be nice to add some newlines to make the pubspec a bit more readable.

Comment (donny-dont) [inline pubspec.yaml]: At this point I was figuring we just fix it up when releasing 0.12.0 into the wild

Comment (nex3) [inline pubspec.yaml]: Sure.

**Pull request merged**